sky/shell/shell.cc

```diff
@@ -68,6 +73,81 @@ void ServiceIsolateHook(bool running_precompiled) {
   }
 }
 
+class JSONResponseBuilder {
+ public:
+  void Append(const char* str) {
+    for (intptr_t i = 0; str[i] != '\0'; i++) {
+      output_.push_back(str[i]);
+    }
+  }
+
+  void Append(const std::string& str) {
+    for (size_t i = 0; i < str.size(); i++) {
+      output_.push_back(str.data()[i]);
+    }
+  }
```

Comment:
`output_.insert(output_.end(), str.data(), str.data() + str.size())` is the usual idom here.